components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1899,6 +1899,42 @@ mod confchange_cmd_metric {
     }
 }
 
+// Validate the request and the split keys
+pub fn validate_and_get_split_keys(
+    split_reqs: &BatchSplitRequest,
+    region_to_split: &Region,
+) -> Result<VecDeque<Vec<u8>>> {
+    if split_reqs.get_requests().is_empty() {
+        return Err(box_err!("missing split requests"));
+    }
+    let mut keys: VecDeque<Vec<u8>> = VecDeque::with_capacity(split_reqs.get_requests().len() + 1);
+    for req in split_reqs.get_requests() {
+        let split_key = req.get_split_key();
+        if split_key.is_empty() {
+            return Err(box_err!("missing split key"));
+        }
+        if split_key
+            <= keys
+                .back()
+                .map_or_else(|| region_to_split.get_start_key(), Vec::as_slice)
+        {
+            return Err(box_err!("invalid split request: {:?}", split_reqs));
+        }
+        if req.get_new_peer_ids().len() != region_to_split.get_peers().len() {
+            return Err(box_err!(
+                "invalid new peer id count, need {:?}, but got {:?}",
+                region_to_split.get_peers(),
+                req.get_new_peer_ids()
+            ));
+        }
+        keys.push_back(split_key.to_vec());
+    }
+
+    util::check_key_in_region(keys.back().unwrap(), region_to_split)?;
```

Comment:
Seems `check_key_in_region_exclusive` makes more sense. Though both functions work the same here.

Reply:
done